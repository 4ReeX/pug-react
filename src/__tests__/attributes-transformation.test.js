import testHelper from './test-helper';

testHelper(__dirname + '/attributes-transformation.input.js');
