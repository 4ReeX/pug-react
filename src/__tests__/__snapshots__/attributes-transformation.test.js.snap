// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript output: transformed source code 1`] = `
"const object = {
  first: \\"one-in-object\\",
  second: \\"two-in-object\\"
};

const array = [\\"arr-one\\", \\"arr-two\\"];

module.exports = (
  <div>
    <div className=\\"one two\\" />
    <div className=\\"one two\\" />
    <div className={100} />
    <div className={array} />
    <div className={object} />
    <div className={[\\"one\\", \\"two\\"]} />
    <div className={[\\"one\\", \\"two\\"].join(\\" \\")} />
    <div className={{ first: true }} />
    <div className=\\"mix one two\\" />
    <div className={\\"mix \\" + 100} />
    <div className={\\"mix \\" + array} />
    <div className={\\"mix \\" + object} />
    <div className={[\\"mix\\", \\"one\\", \\"two\\"]} />
    <div className={\\"mix \\" + [\\"one\\", \\"two\\"].join(\\" \\")} />
    <div className={\\"mix \\" + { first: true }} />
  </div>
);
"
`;

exports[`html output: generated html 1`] = `
<div>
  <div
    className="one two"
  />
  <div
    className="one two"
  />
  <div
    className={100}
  />
  <div
    className={
      Array [
        "arr-one",
        "arr-two",
      ]
    }
  />
  <div
    className={
      Object {
        "first": "one-in-object",
        "second": "two-in-object",
      }
    }
  />
  <div
    className={
      Array [
        "one",
        "two",
      ]
    }
  />
  <div
    className="one two"
  />
  <div
    className={
      Object {
        "first": true,
      }
    }
  />
  <div
    className="mix one two"
  />
  <div
    className="mix 100"
  />
  <div
    className="mix arr-one,arr-two"
  />
  <div
    className="mix [object Object]"
  />
  <div
    className={
      Array [
        "mix",
        "one",
        "two",
      ]
    }
  />
  <div
    className="mix one two"
  />
  <div
    className="mix [object Object]"
  />
</div>
`;

exports[`static html output: static html 1`] = `"<div><div class=\\"one two\\"></div><div class=\\"one two\\"></div><div class=\\"100\\"></div><div class=\\"arr-one,arr-two\\"></div><div class=\\"[object Object]\\"></div><div class=\\"one,two\\"></div><div class=\\"one two\\"></div><div class=\\"[object Object]\\"></div><div class=\\"mix one two\\"></div><div class=\\"mix 100\\"></div><div class=\\"mix arr-one,arr-two\\"></div><div class=\\"mix [object Object]\\"></div><div class=\\"mix,one,two\\"></div><div class=\\"mix one two\\"></div><div class=\\"mix [object Object]\\"></div></div>"`;
